daemon/list.go

```diff
@@ -162,10 +162,7 @@ func (daemon *Daemon) filterByNameIDMatches(view container.View, ctx *listContex
 	cntrs := make([]container.Snapshot, 0, len(matches))
 	for id := range matches {
 		c, err := view.Get(id)
-		if err != nil {
-			return nil, err
-		}
-		if c != nil {
+		if err == nil {
```

Comment:
`view.Get()` returns a "no such container" error if the container wasn't found; https://github.com/moby/moby/pull/33882/files#diff-a41e475815729f1ce9540241ddc3942fR138

Guess we should check for that error here, and return other errors?